### ISSUE TITLE
Make `Lookup` a member of `Root` and collapse it with `DirLookup`

### ISSUE THREAD

**Issue by schomatis** [ipfs/go-mfs#31]
I'm creating a separate issue for this since I'm proposing to remove the exported `DirLookup` function, although I don't seem to find any uses of it outside this repo so it shouldn't break any API.

This should close https://github.com/ipfs/go-ipfs/issues/5085.

https://github.com/ipfs/go-mfs/blob/014cd67bd756eea5bb4b36de9dc04f4e9ced2f4b/ops.go#L174-L185

**Comment by nitishm** [comment]
@schomatis Want me to take a stand at this ?

**Comment by schomatis** [comment]
Yes, this seems like a good issue for you, but let's wait until we can land other PRs to adapt the new MFS to `go-ipfs` (we have done many API breaking changes that are not ported back there, this would be one more, and for you to test it you'd have to adapt `go-ipfs` not only to your patch but also to the other API mismatches we haven't solved yet). Ping me in a week about this one, please.

**Comment by aschmahmann** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
